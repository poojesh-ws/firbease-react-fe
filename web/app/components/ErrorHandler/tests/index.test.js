/**
 *
 * Tests for ErrorHandler
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import ErrorHandler from '../index';

describe('<ErrorHandler />', () => {
  const loading = false;
  const launchListError = 'something_went_wrong';

  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ErrorHandler loading={loading} launchListError={launchListError} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 ErrorHandler component if there is error present', () => {
    const { getAllByTestId } = renderWithIntl(<ErrorHandler loading={loading} launchListError={launchListError} />);
    expect(getAllByTestId('error-message').length).toBe(1);
  });
  it('should not show error when the page is loading', () => {
    const { getAllByTestId } = renderWithIntl(<ErrorHandler loading={true} launchListError={launchListError} />);
    expect(() => getAllByTestId('error-message')).toThrowError();
  });
});
