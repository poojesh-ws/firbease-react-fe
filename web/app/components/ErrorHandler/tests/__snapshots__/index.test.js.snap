// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorHandler /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered ErrorHandler__CustomCard-sc-1rh55w4-0 eOLbNC"
      data-testid="error-card"
    >
      <div
        class="ant-card-body"
      >
        <p
          class="T__StyledText-znbtqz-0 TmgHn"
          data-testid="error-message"
        >
          something_went_wrong
        </p>
      </div>
    </div>
  </div>
</body>
`;
